tests/src/core/testqgsproject.cpp

```diff
@@ -699,5 +700,24 @@ void TestQgsProject::testCrsExpressions()
   QCOMPARE( r.toString(), QString( "EPSG:7030" ) );
 }
 
+void TestQgsProject::testDefaultRelativePaths()
+{
+  QgsSettings s;
+  bool bk_defaultRelativePaths = s.value( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), QVariant( true ) ).toBool();
+
+  s.setValue( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), true );
+  QgsProject p1;
+  bool p1PathsAbsolute = p1.readBoolEntry( QStringLiteral( "Paths" ), QStringLiteral( "/Absolute" ), false );
+
+  s.setValue( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), false );
+  QgsProject p2;
+  bool p2PathsAbsolute = p2.readBoolEntry( QStringLiteral( "Paths" ), QStringLiteral( "/Absolute" ), false );
+
+  s.setValue( QStringLiteral( "/qgis/defaultProjectPathsRelative" ), bk_defaultRelativePaths );
+
```

Comment:
I'd suggest also testing this with an explicit QgsProject::clear call here too, to mimic what qgis app actually does